### PR TITLE
[css-values] Adding new units: en, cap, bs/hs; m/dm/mu, dd/cc, thm/mdl, dot, yd/ft/hd, sx/tx/twip, ax/ay/bx/by/cx/cy; tip/tap/mark/pad/span/sole; moa/soa, pi, perm; min/h, now/mom, beat/blink, jif; bpm

### PR DESCRIPTION
**Pull request by Crissov**
My git-fu isn’t strong enough yet to split this nicely into separate pull requests.

# Length units

## Viewport-relative length units

* `vi` and `vb` logical viewport units. – were already included (#113, #323)

## Relative length units

* Add the [`ic` and] `en` advance units. (#357; #315)
* Add the `cap` unit. (#660, #315, #661)
* Add the `bs` and `hs` stroke units. (#239, #459, #838)

## Absolute length units

* Add the `m`, `dm` and `mu` metric units. (#315)
* Add the `dd` and `cc` metric typographic units. (#315)
* Add the `thm`, `mdl` metric English units.
* Add the `dot` greatest common divisor unit. (#315) 
* Add the `yd`, `ft` and `hd` English units. (#315)
* Add the `sx`, `tx` and `twip` English fractional typographic units. (#315, #378)
* Add the `ax`, `ay`, `bx`, `by`, `cx` and `cy` paper units. (#328, #378)

## Anthropometric/physiological length units 

* Add the `tip`, `tap`, `mark`, `pad`, `span` and `sole` anthropometric units. (#614, #837)

# Angle units

* Add the `moa` and `soa` angular units.
* Add the `pi` circular unit. (#309)
* Add the `perm` slope unit.

# Duration units

* Add the classic `min` and `h` units.
* Add the psychological `now` and `mom` units. (#837)
* Add the physiological (heart) `beat` and (eye) `blink` units. (#837)
* Add the technical `jif` unit.

# Frequency units 

* Add the classic `bpm` unit.

### CODE REVIEW

**Comment by fantasai** [comment]
Please do not issue pull requests for significant changes. These changes need to be discussed and approved by the CSSWG before any text is added to the spec, and pull requests are not an appropriate place for that.

File: css-values/Overview.bs [hunk 1]
```diff
@@ -1274,16 +1549,55 @@ Angle Units: the <<angle>> type and ''deg'', ''grad'', ''rad'', ''turn'' units</
...
+		<dd>Time between user heart beats. 
+			Depends on the user and situation, longer if relaxed.
+			A beat defaults to 0.8s, 
+			i.e. a pulse rate of 75 beats per minute.
+		
+		<dt><dfn id="blink">blink unit</dfn>
+		<dd>Duration the user's eyelid is closed during a spontaneous blink. 
+			Depends on the user and situation, longer if tired. 
+			A blink defaults to 25 milliseconds.
+
```
Comment (Crissov) [inline css-values/Overview.bs]: A new hardware-dependent unit for optimized animations etc., not yet discussed in an Github issue.

File: css-values/Overview.bs [hunk 2]
```diff
@@ -1304,16 +1618,26 @@ Duration Units: the <<time>> type and ''s'', ''ms'' units</h3>
 -->
 
 <h3 id="frequency">
-Frequency Units: the <<frequency>> type and ''Hz'', ''kHz'' units</h3>
```
Comment (Crissov) [inline css-values/Overview.bs]: line breaks for better maintainability, proposed unit added

File: css-values/Overview.bs [hunk 3]
```diff
@@ -1304,16 +1618,26 @@ Duration Units: the <<time>> type and ''s'', ''ms'' units</h3>
...
 	Frequency values are <a>dimensions</a> denoted by <dfn id="frequency-value">&lt;frequency></dfn>.
 	The frequency unit identifiers are:
 
 	<dl export dfn-type=value dfn-for="<frequency>">
-	  <dt><dfn id="Hz">Hz</dfn>
+
+	  <dt><dfn id="Hz">Hz unit</dfn>
 	  <dd>Hertz. It represents the number of occurrences per second.
-	  <dt><dfn id="kHz">kHz</dfn>
-	  <dd>KiloHertz. A kiloHertz is 1000 Hertz.
```
Comment (Crissov) [inline css-values/Overview.bs]: proper spelling in English is lowercase, SI unit symbol is has initial uppercase letter.

File: css-values/Overview.bs [hunk 4]
```diff
@@ -1304,16 +1618,26 @@ Duration Units: the <<time>> type and ''s'', ''ms'' units</h3>
...
+
+	  <dt><dfn id="kHz">kHz unit</dfn>
+	  <dd>Kilohertz. A kilohertz is 1000 hertz. 
+	  	It represents the number of occurrences per millisecond.
+
```
Comment (Crissov) [inline css-values/Overview.bs]: New classic `bpm` unit. May be useful with audio or animation in the future.

File: css-values/Overview.bs [hunk 5]
```diff
@@ -1333,20 +1657,24 @@ Frequency Units: the <<frequency>> type and ''Hz'', ''kHz'' units</h3>
 -->
 
 <h3 id="resolution">
-Resolution Units: the <<resolution>> type and ''dpi'', ''dpcm'', ''dppx'' units</h3>
```
Comment (Crissov) [inline css-values/Overview.bs]: only copyediting

File: css-values/Overview.bs [hunk 6]
```diff
@@ -1893,23 +2230,23 @@ Toggling Between Values: ''toggle()''</h3>
 		margin-left:   toggle(2px, 4px, 5px);
 		</pre>
 
-		Note that, since ''1px'' appears twice in the top and bottom margins,
-		they will cycle between only two values
-		while the left and right margins cycle through three.
```
Comment (Crissov) [inline css-values/Overview.bs]: Example and description didn’t match (any more). Tried my best.

File: css-values/Overview.bs [hunk 7]
```diff
@@ -2182,14 +2519,14 @@ Appendix A: Recommended Minimum Ranges and Precision of Computed Values</h2>
 			        <small>(within the range -100 &lt; <var>x</var> &lt; 100)</small>
 			    <td>(2<sup>17</sup>&minus;1)%
 			<tr><th><<length>>
-			    <td>0.1px
-			    <td>(2<sup>23</sup>&minus;1)px
```
Comment (Crissov) [inline css-values/Overview.bs]: adds relatable minimum min and max sizes

File: css-values/Overview.bs [hunk 8]
```diff
@@ -2273,7 +2609,35 @@ Changes</h2>
 	Changes since <a href="http://www.w3.org/TR/css-values-3/">Level 3</a>:
 
 	<ul>
-		<li>Added the ''vi'', ''vb'', and ''ic'' units.
```
Comment (Crissov) [inline css-values/Overview.bs]: tracking all proposed changes, line by line for maintainability, with Github issue IDs and grouping inside comments

File: css-values/Overview.bs [hunk 9]
```diff
@@ -2287,4 +2651,10 @@ Security and Privacy Considerations</h2>
 	This specification defines units that expose the user's screen size
 	and default font size,
 	but both are trivially observable from JS,
-	so they do not constitutate a new privacy risk.
```
Comment (Crissov) [inline css-values/Overview.bs]: Starting to add new considerations for proposed units. Most of them are just as safe as existing ones.

File: css-values/Overview.bs [hunk 10]
```diff
@@ -1088,27 +1185,93 @@ Absolute lengths: the ''cm'', ''mm'', ''q'', ''in'', ''pt'', ''pc'', ''px'' unit
...
+		<tr><th><dfn id="sx">sx</dfn>
+			<!--"1st" would read "first" -->
+		    <td>sixteenth-inches
+		    <td>1sx = 1in/16 = 6px
+		<tr><th><dfn id="tx">tx</dfn>
+		    <td>thirtysecondth-inches
+		    <td>1tx = 1in/32 = 3px
 		<tr><th><dfn id="pt">pt</dfn>
 		    <td>points
-		    <td>1pt = 1/72th of 1in
```
Comment (Crissov) [inline css-values/Overview.bs]: use notation valid inside `calc()` instead of a mix of English and code notation

File: css-values/Overview.bs [hunk 11]
```diff
@@ -1088,27 +1185,93 @@ Absolute lengths: the ''cm'', ''mm'', ''q'', ''in'', ''pt'', ''pc'', ''px'' unit
...
+		    <td>didots
+		    <td>1dd = 3mm/8 = 1.5q
+		<tr><th><dfn id="thm">thm</dfn>
+			<!--tm, th, thb, tb, nn, ii, inm "metric inch"-->
+		    <td>thumbs
+		    <td>1thm = 25mm
+		<tr><th><dfn id="mdl">mdl</dfn>
+			<!--nn, mod, ftm "metric foot"-->
+		    <td>modules
+		    <td>1mdl = 300mm = 12thm
```
Comment (Crissov) [inline css-values/Overview.bs]: New unit `dot` or `emu` for the greatest common denominator of the smallest units supported by CSS(3). Mentioned in #315.

File: css-values/Overview.bs [hunk 12]
```diff
@@ -1609,11 +1937,12 @@ Type Checking</h4>
 
 		<li>
 			At ''*'',
-			check that at least one side is <<number>>.
```
Comment (Crissov) [inline css-values/Overview.bs]: `<<integer>>` is a subset of `<<number>>`, but mentioning it explicitly should make the next line less surprising.

**Comment by Crissov** [comment]
@fantasai I agree that “these changes need to be discussed and approved by the CSSWG”, but I thought that, since Pull requests are really just a special type of issue on Github, they were the “appropriate place for that”. Editors like you could just remove the parts the WG diapproved of.

**Pull request closed without merging**